### PR TITLE
Unify circleci and build_docs workflows across repositories 

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

Implement of https://github.com/napari/docs/issues/261 in the main repository

# Description

Uniform circleci workflow across main and docs repository for easier maintenance

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,72 +1,48 @@
...
+          environment:
+            PIP_CONSTRAINT: napari/resources/constraints/constraints_py3.10_docs.txt
       - run:
           name: Build docs
           command: |
             . venv/bin/activate
-            cd napari-docs
-            xvfb-run --auto-servernum make docs GALLERY_PATH=../../examples/
+            cd docs
+            xvfb-run --auto-servernum make docs
```
Comment (melissawm) [inline .circleci/config.yml:43]: Did you check to see if this runs? I remember having trouble with the paths where the repos were cloned on CircleCI, which is why I had to explicitly point to the GALLERY_PATH

**Review by melissawm — APPROVED** [review]
Looks good to me in general!

File: .circleci/config.yml (hunk 1 above)
Comment (Czaki) [reply .circleci/config.yml:43]: yes. It works because I have change the checkout path.

**Comment by psobolewskiPhD** [comment]
I kinda would prefer to keep some of the circleCI comments in the workflow?
They should make maintenance a bit easier, given that we don't use those workflows routinely.

**Comment by Czaki** [comment]
> I kinda would prefer to keep some of the circleCI comments in the workflow?
> They should make maintenance a bit easier, given that we don't use those workflows routinely.

Previous comments are from the circle ci template and not was very informative for me. I may add new comments with specific explanations.

**Review by jni — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,72 +1,48 @@
-# Use the latest 2.1 version of CircleCI pipeline process engine.
-# See: https://circleci.com/docs/2.0/configuration-reference
```
Comment (psobolewskiPhD) [inline .circleci/config.yml:2]: Could we keep comments with the useful links?

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,72 +1,48 @@
...
-  # See the orb documentation here: https://circleci.com/developer/orbs/orb/circleci/python
   python: circleci/python@1.5.0
-
-# Define a job to be invoked later in a workflow.
-# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
 jobs:
-  build-docs: # This is the name of the job, feel free to change it to better match what you're trying to do!
-    # These next lines defines a Docker executors: https://circleci.com/docs/2.0/executor-types/
-    # You can specify an image from Dockerhub or use one of the convenience images from CircleCI's Developer Hub
-    # A list of available CircleCI Docker convenience images are available here: https://circleci.com/developer/images/image/cimg/python
```
Comment (psobolewskiPhD) [inline .circleci/config.yml:20]: Same as above, I'd like to keep this link as a comment--we're likely to be changing this line at some points in the future, so why make it harder?

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
I'd really like to at a minimum keep these links in as comments. I think it will help with future maintenance to not have to track this stuff down.
I don't care about the verbiage around the links so much, but I think the links should stay.

**Comment by Czaki** [comment]
@psobolewskiPhD I have restored the most important part of comments with keeping it short to not reduce readability.

But maybe I wrongly select it. So maybe there is a need to restore more.

**Comment by psobolewskiPhD** [comment]
I think those are the important ones.
BTW reading the python orb docs I'm not sure we also need the docker image. Can investigate another PR.

**Review by psobolewskiPhD — APPROVED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,72 +1,53 @@
 # Use the latest 2.1 version of CircleCI pipeline process engine.
```
Comment (psobolewskiPhD) [inline .circleci/config.yml:3]:
```suggestion
# As much as possible, this file should be kept in sync with:
# https://github.com/napari/docs/blob/main/.circleci/config.yaml
# Use the latest 2.1 version of CircleCI pipeline process engine.
```

**Review by melissawm — APPROVED** [review]

**Pull request merged**